lua/entities/gmod_wire_expression2/core/wirelink.lua

```diff
@@ -107,6 +107,17 @@ local function WriteArray(entity, address, data)
 	return free_address
 end
 
+local function writeArraySimple(entity, address, data)
+	local written = 0
+	for index, value in pairs(data) do
+		if type(value) == "number" then
```

Comment:
He's passing the `n` field of the table, so the index doesn't need to be checked